### PR TITLE
Add User-Agent header to OTLP exporter requests

### PR DESCRIPTION
**Pull request by rogerogers**
Signed-off-by: rogerogers <rogers@rogerogers.com>

Implements spec https://github.com/open-telemetry/opentelemetry-specification/pull/2684

#3258

### CODE REVIEW

File: exporters/otlp/otlpmetric/otlpmetricgrpc/client.go [hunk 1]
```diff
@@ -77,6 +78,8 @@ func newClient(ctx context.Context, options ...Option) (otlpmetric.Client, error
 	}
 
 	if c.conn == nil {
+		// Add a User-Agent header when no ClientConn was provided
+		cfg.DialOptions = append(cfg.DialOptions, grpc.WithUserAgent(internal.GetUserAgentHeader()))
```
Comment (dmathieu) [inline exporters/otlp/otlpmetric/otlpmetricgrpc/client.go:82]: If the user of the library is passing their own user agent (Here, in `cf.DialOptions`, for HTTP in `cf.Headers`), the behaviors between HTTP and gRPC differ.

In gRPC, we append the user agent to the passed options. So our value will always take precedence.
In HTTP, we set our header before setting the header options. So the user's value will take precedence.

I think the behavior should be the same for both protocols, and documented/tested.
My preference would go for respecting the user's options. If they wish to override the user-agent, they should be able to do so.

**Review by dmathieu — CHANGES_REQUESTED** [review]
Overall, the change looks good.

File: exporters/otlp/otlpmetric/otlpmetricgrpc/client.go (hunk 1 above)
Comment (rogerogers) [reply exporters/otlp/otlpmetric/otlpmetricgrpc/client.go]: Totally agree with you, move it to init, will allow user override it.

**Review by dmathieu — APPROVED** [review]

**Comment by dmathieu** [comment]
Sorry, I meant to remove my red blocker and GitHub thought that meant I approved the PR 🤦

File: exporters/otlp/otlpmetric/otlpmetricgrpc/client.go (hunk 1 above)
Comment (dmathieu) [reply exporters/otlp/otlpmetric/otlpmetricgrpc/client.go]: Could you add tests as well for this behavior?

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 ### Added
 
 - Added an example of using metric views to customize instruments. (#3177)
+- Add User-Agent header to OTLP exporter requests (#3261)
```
Comment (dmathieu) [inline CHANGELOG.md:14]:
```suggestion
- Add User-Agent header to OTLP exporter requests, if none were set explicitly. (#3261)
```

Comment (rogerogers) [reply CHANGELOG.md]: done, thanks a lot

File: exporters/otlp/otlpmetric/otlpmetricgrpc/client.go (hunk 1 above)
Comment (rogerogers) [reply exporters/otlp/otlpmetric/otlpmetricgrpc/client.go]: ok, I add some tests for this.

**Review by dmathieu — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

**Review by hanyuancheung — APPROVED** [review]

**Pull request merged**